{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/passala/miniconda3/envs/SAMap/lib/python3.7/site-packages/tqdm/auto.py:22: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "from samap.mapping import SAMAP\n",
    "from samap.analysis import (get_mapping_scores, GenePairFinder,\n",
    "                            sankey_plot, chord_plot, CellTypeTriangles, \n",
    "                            ParalogSubstitutions, FunctionalEnrichment,\n",
    "                            convert_eggnog_to_homologs, GeneTriangles)\n",
    "from samalg import SAM\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "fn1 = '/data/passala/Collaborator_Data/Allen_Brain_Data_and_Heart_for_CRUSH_testing/Sam_git_dump/samap_directory/example_data/planarian.h5ad'\n",
    "fn2 = '/data/passala/Collaborator_Data/Allen_Brain_Data_and_Heart_for_CRUSH_testing/Sam_git_dump/samap_directory/example_data/schistosome.h5ad'\n",
    "fn3 = '/data/passala/Collaborator_Data/Allen_Brain_Data_and_Heart_for_CRUSH_testing/Sam_git_dump/samap_directory/example_data/hydra.h5ad'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "filenames = {'pl':fn1,'sc':fn2,'hy':fn3}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "sam1=SAM()\n",
    "sam1.load_data(fn1)\n",
    "\n",
    "sam2=SAM()\n",
    "sam2.load_data(fn2)\n",
    "\n",
    "sam3=SAM()\n",
    "sam3.load_data(fn3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14355 `pl` gene symbols match between the datasets and the BLAST graph.\n",
      "7605 `sc` gene symbols match between the datasets and the BLAST graph.\n",
      "12637 `hy` gene symbols match between the datasets and the BLAST graph.\n"
     ]
    }
   ],
   "source": [
    "sams = {'pl':sam1,'sc':sam2,'hy':sam3}\n",
    "\n",
    "sm = SAMAP(\n",
    "        sams,\n",
    "        f_maps = '/data/passala/Collaborator_Data/Allen_Brain_Data_and_Heart_for_CRUSH_testing/Sam_git_dump/samap_directory/example_data/maps/',\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prepping datasets for translation.\n",
      "Translating feature spaces pairwise.\n",
      "Projecting data into joint latent space. 2.9675638675689697\n",
      "Correcting data with means. 3.5546767711639404\n",
      "Expanding neighbourhoods of species pl...\n",
      "Expanding neighbourhoods of species sc...\n",
      "Expanding neighbourhoods of species hy...\n",
      "Indegree coarsening\n",
      "0/1 (0, 3124)\n",
      "Rescaling edge weights by expression correlations.\n",
      "Concatenating SAM objects...\n",
      "Iteration 1 complete.\n",
      "Alignment scores:\n",
      "          hy        pl        sc\n",
      "hy  0.000000  0.435445  0.424557\n",
      "pl  0.476469  0.000000  0.639517\n",
      "sc  0.478271  0.663831  0.000000\n",
      "Calculating gene-gene correlations in the homology graph...\n",
      "Prepping datasets for translation.\n",
      "Translating feature spaces pairwise.\n",
      "Projecting data into joint latent space. 2.5127639770507812\n",
      "Correcting data with means. 3.0592191219329834\n",
      "Expanding neighbourhoods of species pl...\n",
      "Expanding neighbourhoods of species sc...\n",
      "Expanding neighbourhoods of species hy...\n",
      "Indegree coarsening\n",
      "0/1 (0, 3124)\n",
      "Rescaling edge weights by expression correlations.\n",
      "Concatenating SAM objects...\n",
      "Iteration 2 complete.\n",
      "Alignment scores:\n",
      "          hy        pl        sc\n",
      "hy  0.000000  0.526824  0.488207\n",
      "pl  0.582278  0.000000  0.705608\n",
      "sc  0.555779  0.703633  0.000000\n",
      "Calculating gene-gene correlations in the homology graph...\n",
      "Prepping datasets for translation.\n",
      "Translating feature spaces pairwise.\n",
      "Projecting data into joint latent space. 2.4817702770233154\n",
      "Correcting data with means. 3.1460072994232178\n",
      "Expanding neighbourhoods of species pl...\n",
      "Expanding neighbourhoods of species sc...\n",
      "Expanding neighbourhoods of species hy...\n",
      "Indegree coarsening\n",
      "0/1 (0, 3124)\n",
      "Rescaling edge weights by expression correlations.\n",
      "Concatenating SAM objects...\n",
      "Iteration 3 complete.\n",
      "Alignment scores:\n",
      "          hy        pl        sc\n",
      "hy  0.000000  0.548753  0.518754\n",
      "pl  0.595170  0.000000  0.709267\n",
      "sc  0.570487  0.709469  0.000000\n",
      "Running UMAP on the stitched manifolds.\n",
      "Elapsed time: 1.3182270487149557 minutes.\n"
     ]
    }
   ],
   "source": [
    "sm.run(pairwise=True)\n",
    "samap = sm.samap # SAM object with three species stitched together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "keys = {'pl':'cluster','hy':'Cluster','sc':'tissue'}\n",
    "D,MappingTable = get_mapping_scores(sm,keys,n_top = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"2\" halign=\"left\">pl_Muscle: 13</th>\n",
       "      <th colspan=\"2\" halign=\"left\">sc_Muscle</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pl_Muscle: 14</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pl_Neoblast: 0</th>\n",
       "      <th colspan=\"2\" halign=\"left\">sc_Neoblast</th>\n",
       "      <th>...</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pl_Protonephridia: 26</th>\n",
       "      <th colspan=\"2\" halign=\"left\">sc_Intestine</th>\n",
       "      <th colspan=\"2\" halign=\"left\">pl_Protonephridia: 29</th>\n",
       "      <th colspan=\"2\" halign=\"left\">sc_Gland</th>\n",
       "      <th colspan=\"2\" halign=\"left\">sc_Neural_KK7</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>...</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Alignment score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sc_Muscle</td>\n",
       "      <td>0.981972</td>\n",
       "      <td>pl_Muscle: 13</td>\n",
       "      <td>0.981972</td>\n",
       "      <td>sc_Muscle</td>\n",
       "      <td>0.978676</td>\n",
       "      <td>sc_Neoblast</td>\n",
       "      <td>0.976199</td>\n",
       "      <td>pl_Neoblast: 0</td>\n",
       "      <td>0.976199</td>\n",
       "      <td>...</td>\n",
       "      <td>sc_Parenchymal</td>\n",
       "      <td>0.151563</td>\n",
       "      <td>pl_Cathepsin+ cells: 41</td>\n",
       "      <td>0.147002</td>\n",
       "      <td>sc_Flame cells</td>\n",
       "      <td>0.145002</td>\n",
       "      <td>hy_i_zmg1</td>\n",
       "      <td>0.081615</td>\n",
       "      <td>pl_Neural: 1</td>\n",
       "      <td>0.069167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hy_enEp_tent(pd)</td>\n",
       "      <td>0.440337</td>\n",
       "      <td>pl_Muscle: 14</td>\n",
       "      <td>0.978676</td>\n",
       "      <td>hy_enEp_tent(pd)</td>\n",
       "      <td>0.504562</td>\n",
       "      <td>hy_i_SC</td>\n",
       "      <td>0.656725</td>\n",
       "      <td>pl_Neoblast: 22</td>\n",
       "      <td>0.975029</td>\n",
       "      <td>...</td>\n",
       "      <td>sc_Flame cells</td>\n",
       "      <td>0.035</td>\n",
       "      <td>pl_Cathepsin+ cells: 17</td>\n",
       "      <td>0.111009</td>\n",
       "      <td>hy_ecEp_bat2(mp)</td>\n",
       "      <td>0.066132</td>\n",
       "      <td>pl_Epidermal: 11</td>\n",
       "      <td>0.067572</td>\n",
       "      <td>pl_Neoblast: 0</td>\n",
       "      <td>0.033908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hy_enEp_tent</td>\n",
       "      <td>0.396383</td>\n",
       "      <td>hy_enEp_tent(pd)</td>\n",
       "      <td>0.916387</td>\n",
       "      <td>hy_enEp_tent</td>\n",
       "      <td>0.477911</td>\n",
       "      <td>hy_i_fmgl1</td>\n",
       "      <td>0.654635</td>\n",
       "      <td>hy_i_SC</td>\n",
       "      <td>0.806098</td>\n",
       "      <td>...</td>\n",
       "      <td>sc_Neural</td>\n",
       "      <td>0.017279</td>\n",
       "      <td>hy_enEp_SC2</td>\n",
       "      <td>0.105607</td>\n",
       "      <td>sc_Neural</td>\n",
       "      <td>0.044855</td>\n",
       "      <td>hy_i_zmg2</td>\n",
       "      <td>0.065961</td>\n",
       "      <td>hy_enEp_SC1</td>\n",
       "      <td>0.028247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hy_enEp_foot</td>\n",
       "      <td>0.157747</td>\n",
       "      <td>hy_enEp_tent</td>\n",
       "      <td>0.883429</td>\n",
       "      <td>hy_enEp_foot</td>\n",
       "      <td>0.236074</td>\n",
       "      <td>hy_i_nb1</td>\n",
       "      <td>0.618579</td>\n",
       "      <td>hy_i_nb1</td>\n",
       "      <td>0.804729</td>\n",
       "      <td>...</td>\n",
       "      <td>hy_i_nb3</td>\n",
       "      <td>0.016928</td>\n",
       "      <td>hy_enEp_SC1</td>\n",
       "      <td>0.094195</td>\n",
       "      <td>hy_ecEp_bat1(mp)</td>\n",
       "      <td>0.040683</td>\n",
       "      <td>pl_Parapharyngeal: 12</td>\n",
       "      <td>0.056213</td>\n",
       "      <td>pl_Muscle: 13</td>\n",
       "      <td>0.025964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hy_enEp_nem1(pd)</td>\n",
       "      <td>0.055338</td>\n",
       "      <td>pl_Muscle: 16</td>\n",
       "      <td>0.834026</td>\n",
       "      <td>hy_enEp_nem1(pd)</td>\n",
       "      <td>0.073278</td>\n",
       "      <td>hy_i_nc_gc_prog</td>\n",
       "      <td>0.496306</td>\n",
       "      <td>pl_Neoblast: 5</td>\n",
       "      <td>0.738899</td>\n",
       "      <td>...</td>\n",
       "      <td>hy_i_nb2</td>\n",
       "      <td>0.013348</td>\n",
       "      <td>hy_nan</td>\n",
       "      <td>0.091349</td>\n",
       "      <td>hy_i_nem</td>\n",
       "      <td>0.027394</td>\n",
       "      <td>pl_Epidermal: 3</td>\n",
       "      <td>0.035653</td>\n",
       "      <td>pl_Muscle: 14</td>\n",
       "      <td>0.025434</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 190 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      pl_Muscle: 13                         sc_Muscle                  \\\n",
       "            Cluster Alignment score           Cluster Alignment score   \n",
       "0         sc_Muscle        0.981972     pl_Muscle: 13        0.981972   \n",
       "1  hy_enEp_tent(pd)        0.440337     pl_Muscle: 14        0.978676   \n",
       "2      hy_enEp_tent        0.396383  hy_enEp_tent(pd)        0.916387   \n",
       "3      hy_enEp_foot        0.157747      hy_enEp_tent        0.883429   \n",
       "4  hy_enEp_nem1(pd)        0.055338     pl_Muscle: 16        0.834026   \n",
       "\n",
       "      pl_Muscle: 14                   pl_Neoblast: 0                  \\\n",
       "            Cluster Alignment score          Cluster Alignment score   \n",
       "0         sc_Muscle        0.978676      sc_Neoblast        0.976199   \n",
       "1  hy_enEp_tent(pd)        0.504562          hy_i_SC        0.656725   \n",
       "2      hy_enEp_tent        0.477911       hy_i_fmgl1        0.654635   \n",
       "3      hy_enEp_foot        0.236074         hy_i_nb1        0.618579   \n",
       "4  hy_enEp_nem1(pd)        0.073278  hy_i_nc_gc_prog        0.496306   \n",
       "\n",
       "       sc_Neoblast                  ... pl_Protonephridia: 26                  \\\n",
       "           Cluster Alignment score  ...               Cluster Alignment score   \n",
       "0   pl_Neoblast: 0        0.976199  ...        sc_Parenchymal        0.151563   \n",
       "1  pl_Neoblast: 22        0.975029  ...        sc_Flame cells           0.035   \n",
       "2          hy_i_SC        0.806098  ...             sc_Neural        0.017279   \n",
       "3         hy_i_nb1        0.804729  ...              hy_i_nb3        0.016928   \n",
       "4   pl_Neoblast: 5        0.738899  ...              hy_i_nb2        0.013348   \n",
       "\n",
       "              sc_Intestine                 pl_Protonephridia: 29  \\\n",
       "                   Cluster Alignment score               Cluster   \n",
       "0  pl_Cathepsin+ cells: 41        0.147002        sc_Flame cells   \n",
       "1  pl_Cathepsin+ cells: 17        0.111009      hy_ecEp_bat2(mp)   \n",
       "2              hy_enEp_SC2        0.105607             sc_Neural   \n",
       "3              hy_enEp_SC1        0.094195      hy_ecEp_bat1(mp)   \n",
       "4                   hy_nan        0.091349              hy_i_nem   \n",
       "\n",
       "                                sc_Gland                   sc_Neural_KK7  \\\n",
       "  Alignment score                Cluster Alignment score         Cluster   \n",
       "0        0.145002              hy_i_zmg1        0.081615    pl_Neural: 1   \n",
       "1        0.066132       pl_Epidermal: 11        0.067572  pl_Neoblast: 0   \n",
       "2        0.044855              hy_i_zmg2        0.065961     hy_enEp_SC1   \n",
       "3        0.040683  pl_Parapharyngeal: 12        0.056213   pl_Muscle: 13   \n",
       "4        0.027394        pl_Epidermal: 3        0.035653   pl_Muscle: 14   \n",
       "\n",
       "                   \n",
       "  Alignment score  \n",
       "0        0.069167  \n",
       "1        0.033908  \n",
       "2        0.028247  \n",
       "3        0.025964  \n",
       "4        0.025434  \n",
       "\n",
       "[5 rows x 190 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "D.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SAMap",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
